docs/resources/policy_constraint.md

```diff
@@ -0,0 +1,154 @@
+---
+subcategory: "Beta"
+page_title: "NSXT: nsxt_policy_constraint"
+description: A resource to configure a Constraint (Quota).
+---
+
+# nsxt_policy_constraint
+
+This resource provides a method for the management of a Constraint.
+
+This resource is applicable to NSX Policy Manager.
+
+## Example Usage
+
+```hcl
+resource "nsxt_policy_constraint" "test" {
+  display_name = "demo-quota"
+  description  = "Terraform provisioned Constraint"
+  message      = "too many objects mate"
+
+  target {
+    path_prefix = "/orgs/default/projects/demo/"
+  }
+
+  instance_count {
+    count                = 4
+    target_resource_type = "StaticRoutes"
+  }
+
+  instance_count {
+    count                = 1
+    target_resource_type = "Infra.Tier1.PolicyDnsForwarder"
+  }
+
+  instance_count {
+    count                = 20
+    target_resource_type = "Infra.Domain.Group"
+  }
+}
+```
+
+## Example Usage - Multi-Tenancy
+
+```hcl
+resource "nsxt_policy_constraint" "test" {
+  context {
+    project_id = "demo"
+  }
+
+  display_name = "demo1-quota"
+
+  target {
+    path_prefix = "/orgs/default/projects/demo/vpcs/demo1/"
+  }
+
+  instance_count {
+    count                = 4
+    target_resource_type = "Org.Project.Vpc.PolicyNat.PolicyVpcNatRule"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `context` - (Optional) The context which the object belongs to
+* `display_name` - (Required) Display name of the resource.
+* `description` - (Optional) Description of the resource.
+* `message` - (Optional) User friendly message to be shown to users upon violation.
+* `target` - (Optional) Targets for the constraints to be enforced
+  * `path_prefix` - (Optional) Prefix match to the path, needs to end with `\`
+* `instance_count` - (Optional) Constraint details
+  * `target_resource_type` - (Required) Type of the resource that should be limited in instance count (refer to the table below)
+  * `operator` - (Optional) Either `<=` or `<`. Default is `<=`
+  * `count` - (Required) Limit of instances
+* `tag` - (Optional) A list of scope + tag pairs to associate with this resource.
+* `nsx_id` - (Optional) The NSX ID of this resource. If set, this ID will be used to create the resource.
+
+
+## Attributes Reference
+
+In addition to arguments listed above, the following attributes are exported:
+
+* `id` - ID of the resource.
+* `revision` - Indicates current revision number of the object as seen by NSX-T API server. This attribute can be useful for debugging.
+* `path` - The NSX path of the policy resource.
+
+
+## Target resource types
+
+|Object|project + VPC|project only|VPC only|
```

Comment:
How do we maintain this list? Do we gather this from some doc?

Reply:
This is not documented anywhere as far as I know. I followed the UI to build the table